src/common/pool_map.c

```diff
@@ -1394,7 +1394,6 @@ pool_map_extend(struct pool_map *map, uint32_t version, struct pool_buf *buf)
 	D_DEBUG(DB_TRACE, "Merge buffer with already existent pool map\n");
 	rc = pool_map_merge(map, version, tree);
  out:
-	pool_tree_free(tree);
```

Comment:
Why this is removed? Looks like a memory leak.

Reply:
This was actually a double free. 
_tree_ is freed inside pool_map_merge(...) because it is used to create a temp pool map which is freed inside that function.

Reply:
but there are many error cases (in this pool_map_extend() and pool_map_merge() functions) result in the 'tree' being leaked at the end, right?